core/src/context.hpp

```diff
@@ -23,6 +24,7 @@ namespace core
 class context: public std::list<km_core_context_item>
 {
 public:
+  bool has_markers = true;
```

Comment:
is it really `allow_markers` ?

Would be good to document. `has_markers` sounds like it's only observational.

Reply:
`has` makes sense here because it's about the purpose of the `context` object -- will we be working with markers or not? "My dog has spots" vs "My dog is allowed to have spots" (thanks @rc-swag for the example)